docs/Errors.md

```diff
@@ -48,6 +48,7 @@ Some things to consider in your custom error handler:
 - You can throw a new error in your custom error handler
 	- errors (new error or the received error parameter re-thrown) - will trigger the `onError` lifecycle hook and send the error to the user
 	- an error will not be triggered twice from a lifecycle hook - Fastify internally monitors the error invocation to avoid infinite loops for errors thrown in the reply phases of the lifecycle. (those after the route handler) 
+	- errors thrown by a custom error handler will use an existing headers and status code indicated in the Error object otherwise it will follow the default behavior
```

Comment:
so an error in the error handler, could lead to a 200? - not sure it would be a good thing: in this case the user would miss alert from a monitoring system

In any case, could we add a test for this use case?

Reply:
An error in the error handler will not lead to a 200. It will follow the default behavior as if a custom error handler isn't specified. I believe there was an existing test case for this but I incorrectly changed the test description earlier but I have just changed it back.

Reply:
The sentence is clearer right now 👍 

I searched across the test and I didn't find a test for this use case.
Since there is the new sentence in the readme, could you add the test that verifies that behaviour?

I tried it and I'm not sure it works as expected with

```
fastify.setErrorHandler((err, request, reply) => {
  console.log('error handler')
  throw new Error('ops2')
})

fastify.get('/', async function (request, reply) {
  throw new Error('ops')
})
```

Reply:
I just want to clarify if this test is adequate before I add in a new test.
https://github.com/fastify/fastify/pull/2609/commits/8f9ea6ac91e62d603a180f483f7e21af0374f96d#diff-086d3e4983ab98bad387ce62fb47ff74c0646e94ef5caf4701d096160ea20c06R364-R396

Reply:
That test checks an error sent, I think "errors thrown by a custom error handler" mean a sync error like my example `throw new Error('ops2')` that is an unexpected behaviour by the developer
I would improve this sentence to avoid confusion like I was 😄

Reply:
It should have the same behavior whether it is thrown or sent because some other part of fastify catches it somewhere. I'm not sure where it is tested but I'll improve the sentence.

Reply:
`errors sent or thrown by a custom error handler will be caught safely and routed to Fastify's default error handler for a generic 500 Internal Server Error response.` Is this an adequate description for the behavior?

Reply:
Any thoughts on the current `errors sent or thrown by a custom error handler will be routed to Fastify's default error handler` 
Should I put anything talking about how the reply object will have the default status code / header? I don't think it'd be expected for the reply object to have a status 500 and headers set.